src/posix/platform/rcp_caps_diag.hpp

```diff
@@ -99,11 +99,26 @@ class RcpCapsDiag
         RcpCapsDiag::SpinelCommandHandler mHandler;
     };
 
+    struct FlagEntry
+    {
+        Category mCategory;
+        uint32_t mFlag;
+    };
+
     void ProcessSpinel(void);
+    void ProcessCapabilityFlags(void);
     void TestSpinelCommands(Category aCategory);
+    void TestRadioCapbilityFlags(void);
+    void TestRadioCapbilityFlags(Category aCategory, uint32_t aRadioCapFlags);
+    void TestSpinelCapbilityFlags(void);
+    void TestSpinelCapbilityFlags(Category aCategory, const uint8_t *aCapsData, spinel_size_t aCapsLength);
+    bool IsSpinelCapabilitySupported(const uint8_t *aCapsData, spinel_size_t aCapsLength, uint32_t aCapability);
+    void OutputFormat(const char *aName, const char *aValue);
     void OutputResult(const SpinelEntry &aEntry, otError error);
     void Output(const char *aFormat, ...);
 
+    const char *SupportToString(bool aSupport);
+    const char *RadioCapbilityToString(uint32_t aCapability);
```

Comment:
```suggestion
    static const char *RadioCapbilityToString(uint32_t aCapability);
```
This method can also be declared as static.